packages/@vue/cli-service/bin/postinstall.js

```diff
@@ -0,0 +1,34 @@
+const fs = require('fs')
+const path = require('path')
+
+const stripAnsi = require('strip-ansi')
+const { execa, semver } = require('@vue/cli-shared-utils')
+
+const cwd = process.cwd()
+
+const usesNpm = fs.existsSync(path.resolve(cwd, './package-lock.json'))
```

Comment:
Should check the user agent to see if it was executed by npm, yarn, or pnpm. That also gives you the version currently in use.

```sh
$ env
npm_config_user_agent=npm/6.14.7 node/v15.11.0
npm_config_user_agent=yarn/2.4.0.dev npm/? node/15.11.0
npm_config_user_agent=pnpm/5.4.11 npm/? node/v15.11.0
```

Reply:
Thanks! Didn't know there's a user agent variable available

Reply:
With this environment variable, I think I can move the logic elsewhere. I'll take a try.